frontend/src/metabase/nav/components/AppBar/AppBarSmall.tsx

```diff
@@ -63,34 +67,41 @@ const AppBarSmall = ({
 
   return (
     <AppBarRoot>
-      <AppBarHeader isSubheaderVisible={isSubheaderVisible}>
-        <AppBarMainContainer>
-          <AppBarToggleContainer>
-            {isNavBarEnabled && (
-              <AppBarToggle
-                isNavBarOpen={isNavBarVisible}
-                onToggleClick={onToggleNavbar}
-              />
+      {isHeaderVisible && (
+        <AppBarHeader isSubheaderVisible={isSubheaderVisible}>
+          <AppBarMainContainer>
+            <AppBarToggleContainer>
+              {isNavBarEnabled && (
+                <AppBarToggle
+                  isSmallAppBar
+                  isNavBarOpen={isNavBarVisible}
+                  onToggleClick={onToggleNavbar}
+                />
+              )}
+            </AppBarToggleContainer>
+            <AppBarSearchContainer>
+              {isSearchVisible && (
+                <SearchBar
+                  onSearchActive={handleSearchActive}
+                  onSearchInactive={handleSearchInactive}
+                />
+              )}
+            </AppBarSearchContainer>
+            {isProfileLinkVisible && (
+              <AppBarProfileLinkContainer>
+                <ProfileLink user={currentUser} onLogout={onLogout} />
+              </AppBarProfileLinkContainer>
             )}
-          </AppBarToggleContainer>
-          <AppBarSearchContainer>
-            {isSearchVisible && (
-              <SearchBar
-                onSearchActive={handleSearchActive}
-                onSearchInactive={handleSearchInactive}
-              />
-            )}
-          </AppBarSearchContainer>
-          {isProfileLinkVisible && (
-            <AppBarProfileLinkContainer>
-              <ProfileLink user={currentUser} onLogout={onLogout} />
-            </AppBarProfileLinkContainer>
-          )}
-        </AppBarMainContainer>
-        <AppBarLogoContainer isVisible={!isSearchActive}>
-          <AppBarLogo onLogoClick={handleLogoClick} />
-        </AppBarLogoContainer>
-      </AppBarHeader>
+          </AppBarMainContainer>
+          <AppBarLogoContainer isVisible={isLogoVisible && !isSearchActive}>
+            <AppBarLogo
+              isSmallAppBar
```

Comment:
new prop to determine how to display the component, since we display these components a bit differently between mobile and desktop.